src/java/org/apache/cassandra/config/DataStorage.java

```diff
@@ -0,0 +1,385 @@
+     * Returns the data storage unit.
+     *
+     * @return the data storage unit.
+     */
+    public DataStorageUnit getUnit()
+    {
+        return unit;
+    }
+
+    /**
+     * Returns the amount of data storage in bytes
+     *
+     * @return the amount of data storage in bytes
+     */
+    public long toBytes()
+    {
+        return unit.toBytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in bytes as an {@code int}
+     *
+     * @return the amount of data storage in bytes or {@code Integer.MAX_VALUE} if the number of bytes is too large.
+     */
+    public int toBytesAsInt()
+    {
+        return Ints.saturatedCast(toBytes());
+    }
+
+    /**
+     * Returns the amount of data storage in kilobytes
+     *
+     * @return the amount of data storage in kilobytes
+     */
+    public long toKilobytes()
+    {
+        return unit.toKilobytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in kilobytes as an {@code int}
+     *
+     * @return the amount of data storage in kilobytes or {@code Integer.MAX_VALUE} if the number of kilobytes is too large.
+     */
+    public int toKilobytesAsInt()
+    {
+        return Ints.saturatedCast(toKilobytes());
+    }
+
+    /**
+     * Returns the amount of data storage in megabytes
+     *
+     * @return the amount of data storage in megabytes
+     */
+    public long toMegabytes()
+    {
+        return unit.toMegabytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in megabytes as an {@code int}
+     *
+     * @return the amount of data storage in megabytes or {@code Integer.MAX_VALUE} if the number of megabytes is too large.
+     */
+    public int toMegabytesAsInt()
+    {
+        return Ints.saturatedCast(toMegabytes());
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(unit.toKilobytes(quantity));
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+
+        if (!(obj instanceof DataStorage))
+            return false;
+
+        DataStorage other = (DataStorage) obj;
+        if (unit == other.unit)
+            return quantity == other.quantity;
+
+        // Due to overflows we can only guarantee that the 2 storages are equals if we get the same results
+        // doing the convertion in both directions.
+        return unit.convert(other.quantity, other.unit) == quantity && other.unit.convert(quantity, unit) == other.quantity;
+    }
+
+    @Override
+    public String toString()
+    {
+        return quantity + unit.symbol;
+    }
+
+    public String quantityToString()
+    {
+        return String.valueOf(quantity);
+    }
+
+    public enum DataStorageUnit
+    {
+        BYTES("B")
+        {
+            public long toBytes(long d)
+            {
+                return d;
+            }
+
+            public long toKilobytes(long d)
+            {
+                return d / 1024;
```

Comment:
not consistent with `BitRate`; we use different units in different raveled classes?

1024 = kibi / mebi
1000 = kilo / mega

see https://www.majordifferences.com/2018/03/differences-between-megabyte-and.html

Reply:
I looked back at the ticket again, it was explicitly said to use B/s, KiB/s, MiB/s, GiB/s, TiB/s. for rate units which always refers to 1024kb so this is correct.

Reply:
Looking now into the three parameters using it `compaction_throughput_mb_per_sec` is not clear which one it is according to most people; 
`stream_throughput_outbound` and `inter_dc_stream_throughput_outbound` were explicitly `megabits_per_sec`
Seems to me we need another converter after 4.0 is released with those....? WDYT? 
both @dcapwell @maedhroz

Reply:
I'm guessing `compaction_throughput_mb_per_sec` was meant to be what the article above calls `MiB`. (i.e. It's a usage of MB that informally meant 1024 * 1024 bytes, not 1000 * 1000 bytes.)

Sounds like what we want to do is just use the IEC standard names (kibibyte, mebibyte, gibibyte, etc.) in the method names for this class, change the RegEx to use the IEC abbreviations (KiB, MiB, etc.), and that would bring everything in line with the use of 1024.

Reply:
> stream_throughput_outbound and inter_dc_stream_throughput_outbound were explicitly megabits_per_sec

Right. That would mean we need to interpret any legacy values for those as Mbps and convert to MiB/s.

Reply:
That was a good reference, `CompactionManager#setRate()`, thanks! 
So I will correct that pattern for `DataStorage` and add converter for those two for backward compatibility and everything should fall into places.